packages/-ember-data/tests/integration/store/serializer-for-test.js

```diff
@@ -183,237 +184,268 @@ module('integration/store - serializerFor', function(hooks) {
+          }
         }
-
-        didInit() {
-          appAdapterDidInit = true;
+        class PersonAdapter extends TestAdapter {
+          constructor() {
+            super(...arguments);
+            this.defaultSerializer = null;
+          }
+
+          didInit() {
+            personAdapterDidInit = true;
+          }
         }
-      }
-      class PersonAdapter extends TestAdapter {
-        constructor() {
-          super(...arguments);
-          this.defaultSerializer = null;
+        class FallbackSerializer extends TestSerializer {
+          didInit() {
+            fallbackSerializerDidInit = true;
+          }
         }
-
-        didInit() {
-          personAdapterDidInit = true;
+        class DefaultSerializer extends TestSerializer {
+          didInit() {
+            defaultSerializerDidInit = true;
+          }
         }
+
+        owner.register('adapter:application', AppAdapter);
+        owner.register('adapter:person', PersonAdapter);
+        owner.register('serializer:-fallback', FallbackSerializer);
+        /*
+        serializer:-default is the "last chance" fallback and is
+        registered automatically as the json-api serializer.
+       */
+        owner.unregister('serializer:-default');
+        owner.register('serializer:-default', DefaultSerializer);
+
+        let serializer = store.serializerFor('person');
+
+        assert.ok(serializer instanceof DefaultSerializer, 'We found the serializer');
+        assert.ok(personAdapterDidInit, 'We instantiated the person adapter');
+        assert.ok(!appAdapterDidInit, 'We did not instantiate the application adapter');
+        assert.ok(!fallbackSerializerDidInit, 'We did not instantiate the application adapter fallback serializer');
+        assert.ok(defaultSerializerDidInit, 'We instantiated the `-default` fallback serializer');
+        personAdapterDidInit = false;
+        appAdapterDidInit = false;
+        fallbackSerializerDidInit = false;
+        defaultSerializerDidInit = false;
+
+        let defaultSerializer = store.serializerFor('-default');
+        assert.ok(defaultSerializer instanceof DefaultSerializer, 'We found the correct serializer');
+        assert.ok(!defaultSerializerDidInit, 'We did not instantiate the serializer again');
+        assert.ok(!appAdapterDidInit, 'We did not instantiate the application adapter');
+        assert.ok(!fallbackSerializerDidInit, 'We did not instantiate the application adapter fallback serializer');
+        assert.ok(!personAdapterDidInit, 'We did not instantiate the adapter again');
+        assert.ok(defaultSerializer === serializer, 'We fell back to the fallback-serializer instance');
       }
-      class FallbackSerializer extends TestSerializer {
-        didInit() {
-          fallbackSerializerDidInit = true;
+    );
+
+    deprecatedTest(
+      'specifying defaultSerializer on a fallback serializer when there is no per-type serializer does work',
```

Comment:
specifying defaultSerializer on a fallback `adapter`

Reply:
This was the existing description, was it incorrect before?

Reply:
Yes